### ISSUE TITLE
getopts warning about unhandled option is not emitted with asterisk `*` catch-all clause 

### ISSUE THREAD

**Issue by dasl-** [koalaman/shellcheck#2497]
When the `getopts` case statement has a catch-all `*)` clause, shellcheck will not warn about unhandled options. I understand it may be tricky to detect when the catch-all clause actually handles the option, versus when it is there to detect a usage error. Not sure if shellcheck prefers to err on the side of false positives or false negatives.

#### For new checks and feature suggestions
- [x ] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [ x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
while getopts "vrn" n
do
  case "$n" in
    v) echo "Verbose" ;;
    r) echo "Recursive" ;;
    \?) usage;;
    *) usage;;
  esac
done
```

#### Here's what shellcheck currently says:
Nothing.

#### Here's what I wanted or expected to see:
Similar to https://www.shellcheck.net/wiki/SC2213 : `getopts specified -n, but it’s not handled by this case.`
